app/components/govuk_component/exit_this_page_component.rb

```diff
@@ -0,0 +1,52 @@
+class GovukComponent::ExitThisPageComponent < GovukComponent::Base
+  attr_reader :text, :redirect_url, :secondary
+
+  def initialize(text: config.default_exit_this_page_text, redirect_url: config.default_exit_this_page_redirect_url, secondary: false, classes: [], html_attributes: {})
+    @text = text
+    @redirect_url = redirect_url
+    @secondary = secondary
```

Comment:
I'm not really sure of the best way of naming this. In the design system the section is called ['adding the secondary link'](https://design-system.service.gov.uk/components/exit-this-page/).

Secondary isn't very descriptive though. Would it be better if it was called `skip`? `skip_link`?

Reply:
I think `skip_link`, given that’s what’s used in the code, and the intention of this feature.

Meanwhile, this is annoying: the 2 classes used are inconsistently named:

* `govuk-skip-link`
* `govuk-js-exit-this-page-skiplink`

😵‍💫 I wonder if there’s a naming convention being applied here? Anyway, nothing to do with the project.

Reply:
Yeah, I wondered about just adding an option to the Skip Link helper but that didn't feel quite right. I'll change it to `skip_link`.

Reply:
I'm not sure it makes sense to have the secondary link be part of the same component. Feels to me that it'd be more intuitive to add the option to the existing skip link component? That’s what `govuk-frontend` does too?

Reply:
Ah yes, I see how this works. Yup, I agree with @frankieroberto.

Reply:
That felt unintuitive to me, I'd expect the exit this page functionality to all be in one place.

Reply:
No; the exit this page button appears in a certain location before the page content, where as this second skip link appears after the other skip link, before the header and navigation. So these components would need to be placed independently within a layout.

Reply:
Given these components need to live in separate parts of the page, I’m not sure how else you could have implemented it?

Reply:
I meant from a code organisation point of view, I'd expect the code that generates 'exit this page' HTML to be accessible via `govuk_exit_this_page`, not `govuk_skip_link`.

The upstream implementation of this doesn't feel right:

```
{% block skipLink %}
  {{ super() }}
  {{ govukSkipLink({
    href: "https://bbc.co.uk/weather/",
    classes: "govuk-js-exit-this-page-skiplink"
  }) }}
{% endblock %}
```

Am I skipping to BBC Weather?

I get that it's piggybacking on another component but it feels clunky and less well thought out than the rest of the design system.

Reply:
I prefer the second option.

Also, I'm not hugely keen on the component having a `redirect_url` and the link having a `href`. I might just let the component take either.